### DISCUSSION TITLE
Feature request: Add information about smallest coin value Currency

### DISCUSSION THREAD

**Discussion by OhGhie4iew** [dinerojs/dinero.js#593] [Ideas]
The smallest coin value currently available in SEK is 0.50 SEK.
So in some cases it makes sense to display 1.37 SEK as 1.50 SEK

To produce correct results when rounding it would make sense that the Currency objects also contains information about the smallest coin value.

**Comment by sarahdayan** [comment]
Hey @OhGhie4iew!

Can you walk me through some use cases so I fully understand the problem space?

One of the main goals of Dinero.js is to separate data from its final representation. If you want never to display decimal values other than 0.0 and 0.5, you can do this when formatting with [`toFormat`](https://v2.dinerojs.com/docs/api/formatting/to-format). This way, you preserve the data integrity and accuracy within the object, and respect the reality of SEK when displaying.

```js
import { dinero, toFormat } from 'dinero.js';
import { SEK } from '@dinero.js/currencies';

const d = dinero({ amount: 525, currency: SEK });

toFormat(d, ({ amount, currency }) => {
  if (amount % 0.5 === 0 || amount % 0.5 === 0) {
    return `${currency.code} ${amount}`;
  }

  return // your rounding logic;
});
```

**Comment by sarahdayan** [comment]
Update: `toFormat` has been replaced by [`toDecimal`](https://v2.dinerojs.com/api/formatting/to-decimal.html). You can handle cash rounding in the transformer:

```js
import { dinero, toDecimal } from 'dinero.js';
import { SEK } from 'dinero.js/currencies';

const d = dinero({ amount: 137, currency: SEK });

toDecimal(d, ({ value, currency }) => {
  const num = parseFloat(value);
  const rounded = Math.ceil(num * 2) / 2; // round up to nearest 0.50
  return `${currency.code} ${rounded.toFixed(2)}`;
}); // "SEK 1.50"
```

The principle from my earlier reply still stands: keep the precise value in the Dinero object and apply cash rounding only at display time.
